server/interceptor/interceptor.go

```diff
@@ -107,24 +109,24 @@ func getRPCInfo(req interface{}, fullMethod, quotaUser string) (*rpcInfo, error)
 		return nil, status.Errorf(codes.Internal, "cannot retrieve treeID from request: %T", req)
 	}
 
-	serviceName, methodName, err := parseFullMethod(fullMethod)
-	if err != nil {
-		return nil, err
-	}
-
 	var treeType trillian.TreeType
-	switch serviceName {
-	case "trillian.TrillianAdmin":
+	switch {
+	case isAdminRPC(req):
 		treeType = trillian.TreeType_UNKNOWN_TREE_TYPE // unrestricted
-	case "trillian.TrillianLog":
+	case isLogRPC(req):
 		treeType = trillian.TreeType_LOG
-	case "trillian.TrillianMap":
+	case isMapRPC(req):
 		treeType = trillian.TreeType_MAP
 	default:
-		return nil, status.Errorf(codes.Internal, "cannot determine treeType for request: %T", req)
+		return nil, status.Errorf(codes.Internal, "unmapped request type: %T", req)
 	}
 
-	readonly := strings.HasPrefix(methodName, "Get") || strings.HasPrefix(methodName, "List")
+	reqType := fmt.Sprintf("%T", req)
```

Comment:
OK, fair enough about no reflection if not necessary.

About "someone foolishly choos(ing) the wrong name" for an RPC: I'd be really surprised to have this happen, and even more so if we let the error stand. A non-readonly Get or List method is a significant offence to API style. But, I've taken the comment, so let's leave it at that.